tests/profiling/collector/test_threading.py

```diff
@@ -243,6 +244,13 @@ def play_with_lock():
         pytest.fail("Lock event not found")
 
 
+@pytest.mark.skipif(not TESTING_GEVENT or six.PY3, reason="Not testing gevent on Python 2")
+def test_nogevent_rlock():
```

Comment:
it's great to have this test, but perhaps this test source is not the most appropriate one? Can we make, e.g. `test_nogevent` as part of the internal test suite?